catalog/ui/catalog-ui-search/src/main/webapp/js/cql.spec.js

```diff
@@ -24,5 +24,53 @@ describe('tokenize', () => {
                     expect.fail(0, 1, 'Unexpected filters present');
             }
         });
-    })
+    }),
+
+    describe('filter functions', () => {
+        const cqlString = `proximity('anyText',3,'cat dog')`;
+        const cqlFilter = { 
+            filterFunctionName: 'proximity', 
+            type: 'FILTER_FUNCTION', 
+            params: ['anyText', 3, 'cat dog']
+        };
+
+        it('parses a filter function', () => {
+            const filter = cql.simplify(cql.read(cqlString));
+            expect(filter).to.deep.include(cqlFilter);
+        });
+
+        it('throws when parsing an unknown filter function', () => {
+            const cqlString = `abcdefg('anyText', 3, 'cat dog')`;
+
+            let doParse = function () { cql.read(cqlString); };
+            expect(doParse).to.throw('Unsupported filter function: abcdefg');
+        });
+
+        it('parses a filter function as part of a comparison', () => {
+            const cqlString = `(proximity('anyText',3,'cat dog') = true)`;
+            const cqlOuterFilter = {
+                property: cqlFilter,
+                type: '=',
+                value: true
+            };
+            const filter = cql.simplify(cql.read(cqlString));
+            expect(filter).to.deep.include(cqlOuterFilter);
+        });
+
+        it('parses nested filter functions', () => {
```

Comment:
You get a nested tree of filters back. 
An future example could be:
`
proximity(anyText, 3, strTrim(whatever))
`